opm/io/eclipse/EclFile.cpp

```diff
@@ -648,4 +647,82 @@ EclFile::getImpl(int, eclArrType, const std::unordered_map<int,std::vector<int>>
 template const std::vector<std::string>&
 EclFile::getImpl(int, eclArrType, const std::unordered_map<int,std::vector<std::string>>&, const std::string&);
 
-}} // namespace Opm::EclIO
+}}
+
+
+namespace Opm { namespace EclIO {
+
+void EclFileLGR::initialize_lgr_properties()
+{
+    lgr_array_index.assign(array_name.size(), 0);
+    lgr_array_index_file_entries.assign(array_name.size(), 0);
+
+    int i = 0;
+    for (const auto& entry : array_index) {
+        const std::string& key = entry.first;
+        array_name_to_num[key] = i;
+        i++;
+    }
+
+    for (const auto& entry : array_index) {
+        const std::string& key = entry.first;
+        int key_index = array_name_to_num[key];
```

Comment:
When reading from a `map<>`, you should use `map<>::find()` instead of `map<>::operator[]()` as the latter has the side effect of inserting elements into the map.